tests/sonictool_test.go

```diff
@@ -0,0 +1,409 @@
+
+	net, err := StartIntegrationTestNet(t.TempDir())
+	require.NoError(t, err)
+	for range 2 {
+		createAccount(t, net)
+	}
+	net.Stop()
+
+	_, err = executeSonicTool(t,
+		"--datadir", net.directory+"/state",
+		"compact")
+	require.NoError(t, err)
+}
+
+func TestSonicTool_account_ExecutesWithoutErrors(t *testing.T) {
+	keystoreDir := t.TempDir()
+
+	// Make a password in file (in memory process used by these tests do not allow stdinput rebinding)
+	passwordFileName := t.TempDir() + "/password_file"
+	require.NoError(t, generatePasswordFile(passwordFileName, "this is the passphrase"))
+
+	// no accounts in keystore
+	accounts := listAccounts(t, keystoreDir)
+	require.Empty(t, accounts)
+
+	// Create account
+	accountNewOutput, err := executeSonicTool(t,
+		"--datadir", keystoreDir,
+		"account", "new", "--password", passwordFileName)
+	require.NoError(t, err)
+
+	accounts = listAccounts(t, keystoreDir)
+	require.Len(t, accounts, 1)
+
+	// Parse the address of the created account
+	addressRe := `Public address of the key:\s+([a-zA-Z0-9]+)`
+	matches := regexp.MustCompile(addressRe).FindStringSubmatch(accountNewOutput)
+	require.Len(t, matches, 2)
+	address := matches[1]
+	require.NotEmpty(t, address)
+
+	// Check if the account was created (check if the reported keyfile exists)
+	keyFileRe := fmt.Sprintf("%s/keystore/[^.]+.[^-]+--[a-zA-Z0-9]{40}", keystoreDir)
+	matches = regexp.MustCompile(keyFileRe).FindStringSubmatch(accountNewOutput)
+	require.Len(t, matches, 1)
+	keyfile := matches[0]
+	require.FileExists(t, keyfile)
+
+	// Update account (change password)
+	mockCtrl := gomock.NewController(t)
+	promptMock := prompt.NewMockUserPrompter(mockCtrl)
+	revertPrompt := replaceUserPrompter(promptMock)
+	promptMock.EXPECT().PromptPassword("Passphrase: ").Return("this is the passphrase", nil).Times(2)
+	promptMock.EXPECT().PromptPassword("Repeat passphrase: ").Return("this is the passphrase", nil)
+
+	_, err = executeSonicTool(t,
+		"--datadir", keystoreDir,
+		"account", "update", address)
+	require.NoError(t, err)
+	revertPrompt()
+
+	// Generate a new key to be imported
+	privateKeyFileName := t.TempDir() + "/password_file"
+	_, err = generatePrivateKeyFile(privateKeyFileName)
+	require.NoError(t, err)
+
+	// Import the key
+	_, err = executeSonicTool(t,
+		"--datadir", keystoreDir,
+		"account", "import", privateKeyFileName,
+		"--password", passwordFileName) // key is not encrypted, but we need to provide a password
+	require.NoError(t, err)
+
+	// Check if the account was imported
+	accounts = listAccounts(t, keystoreDir)
+	require.Len(t, accounts, 2)
+}
+
+func TestSonicTool_genesis_ExecutesWithoutErrors(t *testing.T) {
+	tmp := t.TempDir()
+	dataDirA := fmt.Sprintf("%s/A", tmp)
+	dataDirB := fmt.Sprintf("%s/B", tmp)
+
+	// Create a history by running some transactions
+	net, err := StartIntegrationTestNet(dataDirA)
+	require.NoError(t, err)
+	for range 2 {
+		createAccount(t, net)
+	}
+	net.Stop()
```

Comment:
This code fragment is repeated in many tests and seems to mainly serve the purpose to create some client state. How about outlining it into a helper function with a corresponding name?